sycl/test/basic_tests/types.cpp

```diff
@@ -130,8 +142,21 @@ template <typename T> void checkVecUnaryOps(T &v) {
     std::cout << vec2string(g) << std::endl;
   }
 
-  T f = !v;
+  auto f = !v;
   std::cout << vec2string(f) << std::endl;
+
+  // Check operator! return type
+  checkVecNotReturnType<int8_t, int8_t>();
```

Comment:
We are not exactly allowed to do such API changes just yet and therefore the change should be hidden under our new `__INTEL_PREVIEW_BREAKING_CHANGES` macro.

For the implementation itself, that would be a no-op once #11697 is merged, because we will only have to update the new version of the interface. The test, however, will have to be conditionalized and run twice